ethereum/spec/src/test/java/tech/pegasys/teku/spec/logic/common/statetransition/blockvalidator/BatchSignatureVerifierTest.java

```diff
@@ -32,6 +33,32 @@
 
 public class BatchSignatureVerifierTest {
 
+  @Test
+  public void shouldRaiseExceptionIfNoValidPublicKeys() {
+    BatchSignatureVerifier verifier = new BatchSignatureVerifier();
+
+    verifier.verify(
+        List.of(BLSPublicKey.empty()),
+        Bytes.wrap("Hello, world!".getBytes(UTF_8)),
+        BLSTestUtil.randomSignature(43));
+
+    // a case that should probably be handled differently, but equally doesnt seem to occur
+    assertThatThrownBy(verifier::batchVerify).isInstanceOf(IndexOutOfBoundsException.class);
+  }
```

Comment:
This one is happening because even when there are only invalid semiaggregate results `BLS` still does:
```
// completeBatchVerify() call is needed in any case since associated resources could be released
    boolean verifyResult = getBlsImpl().completeBatchVerify(validAggregates);
```

but there is no longer anything to release because BLST stores all memory on the JVM heap now and doesn't use any native memory so we could get rid of this and clean up a bunch of release calls.  Might be better for another PR though.